google/cloud/storage/retry.py

```diff
@@ -37,8 +37,10 @@ def _should_retry(exc):
     """Predicate for determining when to retry."""
     if isinstance(exc, _RETRYABLE_TYPES):
         return True
-    elif isinstance(exc, exceptions.GoogleAPICallError):
+    elif isinstance(exc, api_exceptions.GoogleAPICallError):
         return exc.code in _ADDITIONAL_RETRYABLE_STATUS_CODES
+    elif isinstance(exc, Exception):
+        return _should_retry(exc.args[0])
```

Comment:
This is clever! Just wanted to check--

1. Will exc.args always have at least one item?
2. Can you speak to how you were able to test/repro that this catches the issue in question?

Reply:
So args per Python documentation represents [the tuple](https://docs.python.org/3/library/exceptions.html#BaseException) of constructor parameters:
> The tuple of arguments given to the exception constructor. Some built-in exceptions (like OSError) expect a certain number of arguments and assign a special meaning to the elements of this tuple, while others are usually called only with a single string giving an error message.

My understanding is given this information and the way [auth.TransportError is created](https://github.com/googleapis/google-auth-library-python/blob/f1fee1f4c3d511d9e6ecbc1c0397e743bf2583db/google/auth/transport/requests.py#L188) the underlying Exception should be the first parameter of the construction. I think I could go a step further though and only unwrap when we know it's an auth.TransportError so it's scoped to that specific error.

As for repro, I think the correct thing to do here is add a unit test. What I did was introduce the error in auth package which followed the exception format in issue #414. I'll make a follow-up change.